tailor_distro/manage/compare.py

```diff
@@ -51,19 +52,42 @@ def get_diff(self, repo):
         else:
             diff['name'] = {'unchanged': repo}
 
-        for field in ['type', 'url', 'version']:
-            try:
-                upstream = self.upstream_distro.repositories[repo].source_repository.get_data().get(field, None)
-            except (KeyError, AttributeError):
-                upstream = None
-            try:
-                internal = self.internal_distro.repositories[repo].source_repository.get_data().get(field, None)
-            except (KeyError, AttributeError):
-                internal = None
-
-            if internal != upstream:
-                if internal is not None:
-                    diff[field]['internal'] = internal
-                if upstream is not None:
-                    diff[field]['upstream'] = upstream
+        diff_url = build_diff(
+            upstream=get_url(self.upstream_distro.repositories.get(repo)),
+            internal=get_url(self.internal_distro.repositories.get(repo)))
+
+        if diff_url:
+            diff['url'] = diff_url
+            # version diffs are meaningless when the repository URL is different (e.g. a fork)
+            return diff
+
+        diff_version = build_diff(
+            upstream=get_version(self.upstream_distro.repositories.get(repo)),
+            internal=get_version(self.internal_distro.repositories.get(repo)))
+
+        if diff_version:
+            diff['version'] = diff_version
+
         return diff
+
+
+def get_url(repo):
+    return repo and repo.source_repository and repo.source_repository.url
+
+
+VERSION_TRIM = re.compile("-.+")
+
+
+def get_version(repo):
+    version = repo and repo.release_repository and repo.release_repository.version
+    if version:
+        version = VERSION_TRIM.sub('', version)
```

Comment:
Can you add a comment explaining that this trims off the build number, i.e. 
`# Converts 1.0.0-5 to 1.0.0`